### ISSUE TITLE
SQL injection when encrypting

### ISSUE THREAD

**Issue by davidgecek** [commonsguy/cwac-saferoom#45]
Hi, guys,

there is an issue in the SQLCipherUtils when encrypting the DB with a passphrase containing ' char.  This is the code:

```
 db.rawExecSQL("ATTACH DATABASE '" + newFile.getAbsolutePath() +
                                  "' AS encrypted KEY '" + String.valueOf(passphrase) + "'");
```

Since passphrase is just added as plain text into SQL command, ' in the passphrase closes the String in the SQL and you get something like: SQLiteException: unrecognized token: " "

This is the resulting SQL, as you can see the key is closed as soon as it is opened because of the first char in the passphrase '.
`ATTACH DATABASE '/data/user/0/com.bla.bla/cache/file' AS encrypted KEY ''dahuida'`

I recommend doing something like this:

```
 final SQLiteStatement compiledStatement = db.compileStatement("ATTACH DATABASE ? AS encrypted KEY ?");
            compiledStatement.bindString(1, newFile.getAbsolutePath());
            compiledStatement.bindString(2, String.valueOf(passphrase));
            compiledStatement.execute();
```

I can also make a PR, but I saw that you are more fan of the issue being reported here :)

**Comment by commonsguy** [comment]
I'm not sure that we can use `?` in the first position (for the database path). SQLite does not support placeholders in arbitrary locations in the SQL. The second might work. IIRC, this code was based on some Zetitec example code, and I didn't put much more thought into the process.

I'll give this a try, though it probably won't be for a couple of weeks.

Thanks for pointing this out!

**Comment by davidgecek** [comment]
I'm not sure why this wouldn't work. I tried to make this change locally with both placeholders (both DB path and passphrase) and it seems to work fine...

**Comment by commonsguy** [comment]
Good to know -- thanks!

**Comment by commonsguy** [comment]
Note that the code in your question does not exist in the current version of the project. I do not use `AS encrypted KEY` anywhere. Instead, I use `AS plaintext KEY ''`.

So, I am making your proposed change, to potentially help with strange paths for the database file path, but it will not affect strange characters in the passphrase, which should be working fine already.

**Comment by commonsguy** [comment]
Addressed in `v1.0.1` and `legacy-1.0.1`.

Thanks again!
